### ISSUE TITLE
Bool element always false

### ISSUE THREAD

**Issue by Brandon225** [bluejekyll/foundationdb-rs#150]
all bool elements are being stored as false.

**Comment by Speedy37** [comment]
Could you provide a test case ? There is tests of this in the tuple code, I don’t see how this can happen.

**Comment by Aioxas** [comment]
Noticed here: https://github.com/bluejekyll/foundationdb-rs/blob/0.3/foundationdb/src/tuple/element.rs#L571

There are tests for a straight bool which works as intended, but not for an Element::Bool, going to submit a PR to rectify that

**Comment by Speedy37** [comment]
Nice catch, I didn’t saw it when I checked the code.

By looking at the PR there is another bug in the tuple layer as a tuple at the first level should not be surrounded by NESTED, NIL

**Comment by Aioxas** [comment]
so after the bug fix, it should go as [39, NESTED, 38, NIL] for the given test?

**Comment by Speedy37** [comment]
Yeah

**Comment by Aioxas** [comment]
Figured out encoding, trying to figure out decoding without causing too much hackiness

**Comment by Aioxas** [comment]
Now I'm confused, the documentation says having the tuples encoded as NESTED NIL is okay according to https://github.com/apple/foundationdb/blob/master/design/tuple.md

**Comment by Brandon225** [comment]
My bool test for element inside of `foundationdb/src/tuple/element.rs`  
``` 
 fn test_bool() {
        assert_eq!(
            Element::decode(&[TRUE], TupleDepth(0)).unwrap(),
            (Element::Bool(true), 1)
        );

        assert_eq!(
            Element::decode(&[FALSE], TupleDepth(0)).unwrap(),
            (Element::Bool(false), 1)
        );
    }
}
```

**Comment by Speedy37** [comment]
@Aioxas yeah, but if you look at the official python, go and java bindings, this only apply for nested tuple. At level 0 there is no need for NESTED, ..., NIL. python would decode it as `((...),)`.
This property at level 0 is quite important to get the rest of the "tuple" layer to work (i.e. subspace, directory, versionstamp, ...)

The things is, I can't fix it in 0.3 without it being a breaking change.

**Comment by Aioxas** [comment]
So is this PR still good for merging whenever 0.3.1 is ready, right? While the tuple breaking change will be addressed in a different PR?

**Comment by Speedy37** [comment]
Yeah the PR is still good for merging. I don't know, it's easy to fix, but it would break working code and 0.4 (async/await) is not affected.

Let's open a new issue, and keep this one for the Element::Bool error
